### ISSUE TITLE
Add support for any file type in attachments (image picker button)

### ISSUE THREAD

**Issue by MechDR** [LemmyNet/lemmy-ui#1722]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

Basically, the title. This will make Lemmy a viable alternative for a dying breed of platforms - forums. I run an electronics forum and we exchange a lot of different mime/file types, not just images.

It would at least be nice to have a config somewhere in the settings that lets you change the image upload icon with a general attachment (paperclip) icon and just the UI not get picky about mike types (anything is allowed). Of course, the default should be images.

**Comment by beedell-roke** [comment]
> This will make Lemmy a viable alternative for a dying breed of platforms - forums. I run an electronics forum and we exchange a lot of different mime/file types, not just images.

@MechDR, have you tried Discourse?

**Comment by dessalines** [comment]
Lemmy is not a file-sharing service. Use torrent magnet links.

**Comment by beedell-roke** [comment]
@dessalines, the use case of uploading images, which Lemmy doesn't support rendering, exists. 🤷‍♂️ Regardless, the Blender Discourse instance demonstrates that allowing administrator-specified MIME types appears conceptually reasonable (irrespective of implementation feasibility).

**Comment by dessalines** [comment]
Lemmy stores no pictures, we use a service called pict-rs that runs alongside lemmy for that. 

Regardless uploads are a massive burden for server runners. Use torrents magnet links for sharing data.

**Comment by beedell-roke** [comment]
@dessalines, I was referring to servers where a specific file type is inherent to their operation (like Blender's Discourse instance's `.blend` uploads). Additionally, torrents have a high burden of entry – they're not intuitive for most, else they'd utilised more frequently.

Though, since you state that all attachment storage is external, this is moot, and that's ideal.
